internals/secrethub/generate.go

```diff
@@ -36,13 +43,15 @@ func NewGenerateSecretCommand(io ui.IO, newClient newClientFunc) *GenerateSecret
 
 // Register registers the command, arguments and flags on the provided Registerer.
 func (cmd *GenerateSecretCommand) Register(r Registerer) {
-	generateCommand := r.Command("generate", "").Hidden()
-
-	clause := generateCommand.Command("rand", "Generate a random secret. By default, it uses numbers (0-9), lowercase letters (a-z) and uppercase letters (A-Z) and a length of 22.")
-	clause.Arg("secret-path", "The path to write the generated secret to (<namespace>/<repo>[/<dir>]/<secret>)").Required().SetValue(&cmd.path)
-	clause.Arg("length", "The length of the generated secret. Defaults to 22.").Default("22").IntVar(&cmd.length)
+	clause := r.Command("generate", "Generate a random secret.")
+	clause.HelpLong("By default, it uses numbers (0-9), lowercase letters (a-z) and uppercase letters (A-Z) and a length of 22.")
+	clause.Arg("secret-path", "The path to write the generated secret to (<namespace>/<repo>[/<dir>]/<secret>)").Required().StringVar(&cmd.firstArg)
+	clause.Flag("length", "The length of the generated secret. Defaults to "+strconv.Itoa(defaultLength)).PlaceHolder(strconv.Itoa(defaultLength)).SetValue(&cmd.lengthFlag)
 	clause.Flag("symbols", "Include symbols in secret.").Short('s').BoolVar(&cmd.useSymbols)
 
+	clause.Arg("rand-command", "").Hidden().StringVar(&cmd.secondArg)
```

Comment:
One thing I'm missing is: how are environment variables handled in a backwards compatible way? And, did you 'add' more environment variables that `secrethub printenv -v` will print now?

I'd suggest to only keep `SECRETHUB_GENERATE_RAND_XXX` for backwards compatibility and always have `SECRETHUB_GENERATE_XXX` take precedence.

Reply:
Good catch! For hidden flags an environment variable is still registered, but not for renamed commands. I've added a commit to read the envvar in the command.